bin/fin

```diff
@@ -1716,27 +1716,38 @@ EOF
 }
 
 # Method to create the user and SMB network share on Windows.
-# @param share_name, path
+# @param share_name, share_path
 smb_share_add()
 {
+	local share_name=$1
+	local share_path=$2
+
 	# Add SMB share if it does not exist.
-	if [[ "$(net share)" != *"${1}"*"Docksal ${2} drive"* ]];
+	if [[ "$(net share)" != *"${1}"*"Docksal ${share_path} drive"* ]];
 	then
-		command_share="net share ${1}=${2} /grant:${USERNAME},FULL /REMARK:\"Docksal ${2} drive\""
+		command_share="net share ${share_name}=${share_path} /grant:${USERNAME},FULL /REMARK:\"Docksal ${share_path} drive\""
 		echo-green "Adding docker SMB share..."
 		winsudo "$command_share"
 	fi
 }
 
 # Method to mount the SMB share in Docker machine.
-# @param share_path, mount_point, password
+# @param share_name, mount_point, password
 smb_share_mount()
 {
+	local share_name=$1
+	local mount_point=$2
+	local password=$3
+
 	#DOCKER_MACHINE_IP=$(docker-machine ip ${DOCKER_MACHINE_NAME})
 	network_id=$("$vboxmanage" showvminfo ${DOCKER_MACHINE_NAME} --machinereadable | grep hostonlyadapter | cut -d'"' -f2)
 	DOCKER_HOST_IP=$("$vboxmanage" list hostonlyifs | grep "${network_id}$" -A 3 | grep IPAddress |cut -d ':' -f2 | xargs)
 
-	command_mount="sudo mkdir -p $2 && sudo mount -t cifs -o username='${USERNAME}',pass='$3',sec=ntlm,nobrl,mfsymlinks,noperm,actimeo=1,dir_mode=0777,file_mode=0777 //${DOCKER_HOST_IP}/$1 $2"
+	# Doing unmount first to make it possible to rerun this command without restarting the VM (e.g. when debugging)
+	command_mount="
+		(sudo umount $mount_point >/dev/null 2>&1 || true) && \
```

Comment:
1. Please don't use `&& \` just use `&&`
2. If you want next command to execute anyways instead of using `|| true) &&` just use `;`
```
  sudo umount $mount_point >/dev/null 2>&1;
  sudo mkdir -p $mount_point &&
  sudo mount -t cifs -o ...
```